src/services/milestones/milestones.hooks.js

```diff
@@ -61,6 +62,107 @@ const restrict = () => (context) => {
     .then(milestones => ((Array.isArray(milestones)) ? milestones.forEach(canUpdate) : canUpdate(milestones)));
 };
 
+
+/**
+ *
+ * Conditionally sends a notification after patch or create
+ *
+ **/
+const sendNotification = () => (context) => {
+  const { id, data, app, result } = context;
+
+  /**
+   * Notifications when a milestone get created
+   **/
+  if(context.method === 'create') {
+    if(result.status === 'proposed') {
+      // find the campaign admin and send a notification that milestone is proposed
+      app.service('users').find({query: { address: data.campaignOwnerAddress }})
+        .then((users) => {
+          console.log('campaignOwner', users)
+
+          Notifications.milestoneProposed(app, {
+            recipient: users.data[0].email,
+            user: users.data[0].name,
+            milestoneTitle: data.title,
+            campaignTitle: result.campaign.title,
+            amount: data.maxAmount
+          });  
+        })
+        .catch((e) => console.error('error sending proposed milestone notification', e));
+    };  
+  }
+
+  /**
+   * Notifications when a milestone get patches
+   **/
+  if(context.method === 'patch') {
+    if(result.prevStatus === 'proposed' && result.status === 'InProgress' && result.mined) {
```

Comment:
instead of using `prevStatus` could we just fetch the milestone and check? This would save us from having to remember to set the `prevStatus` in the dapp. One less thing to break.

Reply:
Honestly I couldn't figure out the flow of status and mined, and failed tx 100%, so I didn't want to break things by introducing new states. Introducing prevStats was the easiest solution

Reply:
actually `prevStatus` is needed... but your "mined" comment reminds me that we should only send the notification for after the tx is mined... 

so the `milestoneMarkedCompleted` and `milestoneCanceled` notifications below need a `result.mined` check

Reply:
a potential solution to make the `prevStatus` a bit more robust is to create a feathers hook that sets the `prevStatus` on the milestone for every patch. This would eliminate having to do it in the dapp

Reply:
Can we do these optimizations in the next iteration? I'd like to have the notifications out fast